lib/bashcov/xtrace.rb

```diff
@@ -55,6 +55,21 @@ def read
         @files[filename] ||= []
         @files[filename][lineno] ||= 0
         @files[filename][lineno] += 1
+        line = File.readlines(filename)[lineno]
+        ishere = line.match("cat.*<<-?\s*[\"']?\(.*\)[\"']?")
```

Comment:
You can use heredoc with something else than `cat` (see http://stackoverflow.com/questions/1167746/how-to-assign-a-heredoc-value-to-a-variable-in-bash for example) but I like that idea, it just needs to be more generic.